internal/dispatch/dispatch.go

```diff
@@ -121,5 +122,13 @@ func ExpandRequestToKey(req *v1.DispatchExpandRequest) string {
 
 // ReachableResourcesRequestToKey converts a reachable resources request into a cache key
 func ReachableResourcesRequestToKey(req *v1.DispatchReachableResourcesRequest) string {
-	return fmt.Sprintf("%s//%s#%s@%s@%s", reachableResourcesPrefix, req.ObjectRelation.Namespace, req.ObjectRelation.Relation, tuple.StringONR(req.Subject), req.Metadata.AtRevision)
+	return fmt.Sprintf("%s//%s#%s@%s#%s:[%s]@%s",
+		reachableResourcesPrefix,
+		req.ResourceRelation.Namespace,
+		req.ResourceRelation.Relation,
+		req.SubjectRelation.Namespace,
+		req.SubjectRelation.Relation,
+		strings.Join(req.SubjectIds, ","),
```

Comment:
At standup I proposed changing this to a hash + count, but now I'm thinking we should probably just make sure that the cache memory usage takes into account the fully serialized cache key length. Correctness is more important than performance, and we can't rely on a fast-enough hash to be collision-free.

Reply:
Yeah, the cache size estimator is adding the ID lengths together, so it should be (roughly) correct in its estimate